glfw/src/Joystick.zig

```diff
@@ -407,10 +413,13 @@ pub inline fn isGamepad(self: Joystick) bool {
 /// @thread_safety This function must only be called from the main thread.
 ///
 /// see also: gamepad, glfw.Joystick.isGamepad
-pub inline fn getGamepadName(self: Joystick) Error!?[*c]const u8 {
-    const name = c.glfwGetGamepadName(self.jid);
+pub inline fn getGamepadName(self: Joystick) Error!?[:0]const u8 {
+    const name_opt = c.glfwGetGamepadName(self.jid);
     try getError();
-    return name;
+    return if (name_opt) |name|
+        std.mem.span(name)
+    else
+        null;
```

Comment:
If I understand correctly, `std.mem.span` handles optional pointer types already so we may not need these null checks? https://github.com/ziglang/zig/blob/33889e1974137a54d1a88b87f57462de2a7cc145/lib/std/mem.zig#L623-L629

Reply:
Good point, I should've checked that :)

Reply:
Ah, I remember why this was. `*c` pointers are not optional types, so span won't null-check them

Reply:
Ahh got it!